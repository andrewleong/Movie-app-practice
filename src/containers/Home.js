import React, { Component } from 'react';

// Container imports
import SearchBar from './SearchBar';

// Component imports
import MovieList from '../components/MovieList';


const API_SITE = process.env.REACT_APP_API_SITE;
const API_KEY = process.env.REACT_APP_API_KEY;

class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            movies: [],
        }
    }

    handleSearch = (query) => {
        fetch(`${API_SITE}search/movie?api_key=${API_KEY}&query=${query}`)
        .then((response) => response.json())
        .then(({ results }) => {
            if(results.length) {
                this.setState({
                    movies: results
                })
            }
        });
    }

    render() {
        const { movies } = this.state;
        return (
            <div className="container">
                <SearchBar
                    onSearch={this.handleSearch}
                />

                {
                    movies.length === 0 ?
                        <h1>Please do a search.</h1> :
                        <MovieList
                            movieList={movies}
                        />
                }


            </div>
        );
    }
}


export default Home;
