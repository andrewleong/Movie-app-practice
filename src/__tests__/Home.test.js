import React from 'react';
import { shallow } from 'enzyme';
import Home from '../containers/Home';

const wrapper = shallow(<Home />);

describe('(Container) Home', () => {
    it('renders without crashing', () => {
        expect(wrapper).toHaveLength(1);
    });

    it('should render a <SearchBar />', () => {
        expect(wrapper.find('SearchBar')).toHaveLength(1);
        expect(wrapper.props()).toMatchSnapshot();
    });

    it('should render a <MovieList /> if movies are found', () => {
        const movies = [
            {
                id: 1,
                name: "movie 1"
            }
        ]
        wrapper.setState({ movies })
        expect(wrapper.find('MovieList')).toHaveLength(1);
        expect(wrapper.props()).toMatchSnapshot();
    });

    it('should display a message if there are no movies', () => {
        const movies = []
        wrapper.setState({ movies })
        expect(wrapper.find('h1').text()).toEqual("Please do a search.");
        expect(wrapper.props()).toMatchSnapshot();
    });

});
