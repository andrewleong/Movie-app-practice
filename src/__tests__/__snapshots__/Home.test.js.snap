// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) Home should display a message if there are no movies 1`] = `
Object {
  "children": Array [
    <SearchBar
      onSearch={[Function]}
    />,
    <h1>
      Please do a search.
    </h1>,
  ],
  "className": "container",
}
`;

exports[`(Container) Home should render a <MovieList /> if movies are found 1`] = `
Object {
  "children": Array [
    <SearchBar
      onSearch={[Function]}
    />,
    <MovieList
      movieList={
        Array [
          Object {
            "id": 1,
            "name": "movie 1",
          },
        ]
      }
    />,
  ],
  "className": "container",
}
`;

exports[`(Container) Home should render a <SearchBar /> 1`] = `
Object {
  "children": Array [
    <SearchBar
      onSearch={[Function]}
    />,
    <h1>
      Please do a search.
    </h1>,
  ],
  "className": "container",
}
`;
